gravatar/src/main/java/com/gravatar/services/AvatarService.kt

```diff
@@ -2,71 +2,51 @@ package com.gravatar.services
 import com.gravatar.types.Email
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.launch
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 import okhttp3.MultipartBody
 import okhttp3.OkHttpClient
 import okhttp3.RequestBody.Companion.asRequestBody
-import okhttp3.ResponseBody
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
 import java.io.File
 import com.gravatar.di.container.GravatarSdkContainer.Companion.instance as GravatarSdkDI
 
 /**
  * Service for managing Gravatar avatars.
  */
-public class AvatarService(private val okHttpClient: OkHttpClient? = null) {
+public class AvatarService(okHttpClient: OkHttpClient? = null) {
     private companion object {
         const val LOG_TAG = "AvatarService"
     }
 
-    // Run onResponse and onError callbacks on the main thread
-    private val coroutineScope = CoroutineScope(GravatarSdkDI.dispatcherMain)
+    private val service = GravatarSdkDI.getGravatarApiV1Service(okHttpClient)
 
     /**
      * Uploads a Gravatar image for the given email address.
      *
      * @param file The image file to upload
      * @param email The email address to associate the image with
      * @param wordpressBearerToken The bearer token for the user's WordPress/Gravatar account
-     * @param gravatarUploadListener The listener to notify of the upload result
      */
-    public fun upload(
-        file: File,
-        email: Email,
-        wordpressBearerToken: String,
-        gravatarUploadListener: GravatarListener<Unit, ErrorType>,
-    ) {
-        val service = GravatarSdkDI.getGravatarApiV1Service(okHttpClient)
+    public suspend fun upload(file: File, email: Email, wordpressBearerToken: String): Result<Unit, ErrorType> {
         val identity = MultipartBody.Part.createFormData("account", email.toString())
         val filePart =
             MultipartBody.Part.createFormData("filedata", file.name, file.asRequestBody())
 
-        service.uploadImage("Bearer $wordpressBearerToken", identity, filePart).enqueue(
-            object : Callback<ResponseBody> {
-                override fun onResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
-                    coroutineScope.launch {
-                        if (response.isSuccessful) {
-                            gravatarUploadListener.onSuccess(Unit)
-                        } else {
-                            // Log the response body for debugging purposes if the response is not successful
-                            Logger.w(
-                                LOG_TAG,
-                                "Network call unsuccessful trying to upload Gravatar: $response.body",
-                            )
-                            gravatarUploadListener.onError(errorTypeFromHttpCode(response.code()))
-                        }
-                    }
-                }
+        return runCatchingService {
+            withContext(Dispatchers.IO) {
```

Comment:
For consistency perhaps use this one? `GravatarSdkDI.dispatcherIO` as it seems to be the one managed by us.

Reply:
Good catch! Thanks.